crypto/armcap.c

```diff
@@ -54,6 +55,32 @@ void _armv8_pmull_probe(void);
 # ifdef __aarch64__
 void _armv8_sha512_probe(void);
 unsigned int _armv8_cpuid_probe(void);
+static void _armv8_rng_probe(void);
+
+static size_t OPENSSL_rndr_asm(unsigned char *buf, size_t len);
+static size_t OPENSSL_rndrrs_asm(unsigned char *buf, size_t len);
+
+static size_t OPENSSL_rndr_wrapper(size_t (*func)(unsigned char *, size_t), unsigned char *buf, size_t len)
+{
+    size_t buffer_size;
+    int i;
+
+    for (i = 0; i < 8; i++) {
+	buffer_size = func(buf, len);
+	if (buffer_size == len)
+	    break;
+	usleep(5000);  //5000 microseconds (5 milliseconds)
```

Comment:
Is this an arbitrary decision here for the time?

You also want c style comments.. /* */

Reply:
This was decided upon after some discussion with an engineer colleague who has experience with Arm RNG. 

The spec states in case of failure: “If the instruction cannot return a genuine random number in a reasonable period of time, PSTATE.NZCV is set to 0b0100 and the data value returned is 0.” ([Doc](https://developer.arm.com/documentation/ddi0595/2021-03/AArch64-Registers/RNDR--Random-Number?lang=en)) The engineer I spoke with specified that a “reasonable period of time” is IMPLEMENTATION DEFINED and that true RNGs can go out of spec and need some sort of localized reset. In these types of systems the resetting can take an **order of milliseconds to complete**.

Thus 5 ms was chosen to be used as a delay to allow the system to perform a localized reset in the unlikely event that the rng has gone out of spec/failed. This amount of time was chosen to be in the relative range of how long it would take to perform a localized reset of a true rng system. 3 ms/8ms, or another similarly sized amount of time, would have worked just as well, since this will be run in a loop 8 times.

Reply:
Ok. please fix the c++ style comment

Reply:
Apologies for missing the second comment regarding c++ style comments. Now Fixed.